### DISCUSSION TITLE
Password Grant Deprecated

### DISCUSSION THREAD

**Discussion by jleonardolemos** [laravel/framework#53755] [Q&A]
Password Grant was deprecated at 2.1 version of the OAuth spec. I am i little concerned because i have too much first party apps using Laravel Passport Password Grand as the main authentication option.

Wondering if in the future this will be deprecated for Passport too or if i can continue using Passport for ever.

OBS: I know sanctum is around but i just prefer Passport.

**Comment by hafezdivandari** [comment]
OAuth 2.1 is still in draft, which means the password grant is not officially deprecated yet. It will likely take years for OAuth 2.1 to be stable, and possibly a decade for `league\oauth2-server` to support it. Even then, the password grant will only be deprecated, not removed. You might expect its removal in OAuth 3.0, followed by the eventual removal of support for it in `league\oauth2-server` and Passport. I’d say you’re safe until the end of this century!
